src/org/loklak/harvester/strategy/PriorityKaizenHarvester.java

```diff
@@ -0,0 +1,68 @@
+package org.loklak.harvester.strategy;
+
+import org.loklak.data.DAO;
+
+import java.util.Comparator;
+import java.util.PriorityQueue;
+import java.util.Queue;
+
+public class PriorityKaizenHarvester extends KaizenHarvester {
+
+    private static class PriorityKaizenQueries extends KaizenQueries {
+
+        private Comparator<ScoreWrapper> scoreComparator = (scoreWrapper, t1) -> (int) (scoreWrapper.score - t1.score);
+
+        private Queue<ScoreWrapper> queue;
+        private int maxSize;
+
+        private class ScoreWrapper {
+
+            private double score;
+            private String query;
+
+            ScoreWrapper(String m, double score) {
+                this.query = m;
+                this.score = score;
+            }
+
+        }
+
+        public PriorityKaizenQueries(int size) {
+            this.maxSize = size;
+            queue = new PriorityQueue<>(size, scoreComparator);
```

Comment:
Okay, so here the queries will get dropped only if we run out of memory, This seems good 👍